cmd/dlt/accountroles/cmd.go

```diff
@@ -0,0 +1,224 @@
+		if err != nil {
+			reporter.Errorf("Failed to close OCM connection: %v", err)
+		}
+	}()
+
+	env, err := ocm.GetEnv()
+	if err != nil {
+		reporter.Errorf("Error getting environment %s", err)
+		os.Exit(1)
+	}
+	creator, err := awsClient.GetCreator()
+	if err != nil {
+		reporter.Errorf("Failed to get IAM credentials: %s", err)
+		os.Exit(1)
+	}
+	clusters, err := ocmClient.GetClusters(creator, 1000)
+	if err != nil {
+		reporter.Errorf("Error getting clusters %s", err)
+		os.Exit(1)
+	}
+	finalRoleList := []string{}
+	if args.roleName == "" {
+		roles, err := awsClient.GetAccountRolesForCurrentEnv(env, creator.AccountID)
+		if err != nil {
+			reporter.Errorf("Error getting roles: %s", err)
+			os.Exit(1)
+		}
+		for _, role := range roles {
+			if !checkIfRoleExists(clusters, role) {
+				finalRoleList = append(finalRoleList, role.RoleName)
+			}
+		}
+		if len(finalRoleList) == 0 {
+			reporter.Errorf("There are no roles to be deleted")
+			os.Exit(1)
+		}
+	} else {
+		role, err := awsClient.GetAccountRoleForCurrentEnv(env, args.roleName)
+		if err != nil {
+			reporter.Errorf("Error getting role: %s", err)
+			os.Exit(1)
+		}
+		finalRoleList = append(finalRoleList, role)
+	}
+
+	// Determine if interactive mode is needed
+	if !interactive.Enabled() && !cmd.Flags().Changed("mode") {
+		interactive.Enable()
+	}
+	mode := args.mode
+	if interactive.Enabled() {
+		mode, err = interactive.GetOption(interactive.Input{
+			Question: "Account role deletion mode",
+			Help:     cmd.Flags().Lookup("mode").Usage,
+			Default:  mode,
+			Options:  modes,
+			Required: true,
+		})
+		if err != nil {
+			reporter.Errorf("Expected a valid Account role deletion mode: %s", err)
+			os.Exit(1)
+		}
+	}
+	switch mode {
+	case "auto":
+		ocmClient.LogEvent("ROSADeleteAccountRoleModeAuto")
+		for _, role := range finalRoleList {
+			if !confirm.Prompt(true, "Delete the account role '%s'?", role) {
+				continue
+			}
+			err := awsClient.DeleteAccountRole(role)
+			if err != nil {
+				reporter.Errorf("There was an error deleting the OIDC provider: %s", err)
+				continue
+			}
+		}
+	case "manual":
+		ocmClient.LogEvent("ROSADeleteAccountRoleModeManual")
+		policyMap, err := awsClient.GetAccountRolePolicies(finalRoleList)
+		if err != nil {
+			reporter.Errorf("There was an error getting the policy: %v", err)
+			os.Exit(1)
+		}
+		commands := buildCommand(finalRoleList, policyMap)
+		if reporter.IsTerminal() {
+			reporter.Infof("Run the following commands to delete the Operator roles:\n")
+		}
+		fmt.Println(commands)
+	}
+}
+
+func checkIfRoleExists(clusters []*cmv1.Cluster, role aws.Role) bool {
+	for _, cluster := range clusters {
+		if cluster.AWS().STS().RoleARN() == role.RoleARN ||
+			cluster.AWS().STS().SupportRoleARN() == role.RoleARN ||
+			cluster.AWS().STS().InstanceIAMRoles().MasterRoleARN() == role.RoleARN ||
+			cluster.AWS().STS().InstanceIAMRoles().WorkerRoleARN() == role.RoleARN {
+			return true
+		}
+	}
+	return false
+}
```

Comment:
I suspect we can (and should) do this in a filter to the OCM API.

```
ocm get clusters --parameter search="aws.sts.role_arn = 'arn:aws:iam::$(creator.AccountID):role/$(roleName)' OR aws.sts.support_role_arn = 'arn:aws:iam::$(creator.AccountID):role/$(roleName)' OR etc..."
```
Alternatively, we can optionally take _just_ the prefix (as a separate, optional, and mutually exclusive flag) and delete all roles that match the prefix.